sketches/src/main/java/org/apache/apex/malhar/sketches/QuantilesEstimator.java

```diff
@@ -0,0 +1,182 @@
+ * quantiles of the probability distribution. Leverages the quantiles sketch implementation from the Yahoo Datasketches
+ * Library.
+ * <p/>
+ * <b>Input Port(s) : </b><br>
+ * <b>data : </b> Data values input port. <br>
+ * <br>
+ * <b>Output Port(s) : </b> <br>
+ * <b>cdfOutput : </b>cumulative distribution function output port. <br>
+ * <b>quantilesOutput : </b>quantiles output port. <br>
+ * <br>
+ * <b>Partitions : No</b>, no will yield wrong results. <br>
+ * <br>+
+ */
+@OperatorAnnotation(partitionable = false)
+public class QuantilesEstimator extends BaseOperator
+{
+  /**
+   * Output port that emits cdf estimated at the current data point
+   */
+  public final transient DefaultOutputPort<Double> cdfOutput = new DefaultOutputPort<>();
+  /**
+   * Emits quantiles of stream seen thus far
+   */
+  public final transient DefaultOutputPort<double[]> quantilesOutput = new DefaultOutputPort<>();
+  /**
+   * Emits probability masses on specified intervals
+   */
+  public final transient DefaultOutputPort<double[]> pmfOutput = new DefaultOutputPort<>();
+  private transient QuantilesSketch quantilesSketch = QuantilesSketch.builder().build();
+  /**
+   * This field determines the specific quantiles to be calculated.
+   * Default is set to compute the standard quartiles.
+   */
+  private double[] fractions = {0.0, 0.25, 0.50, 0.75, 1.00};
+  /**
+   * This field determines the intervals on which the probability mass function is computed.
+   */
+  private double[] pmfIntervals = {};
+  /**
+   * This operator computes three different quantities which are output on separate output ports. If not using any of
+   * these quantities, these variables can be set to avoid unnecessary computation.
+   */
+  private boolean computeCdf = true;
+  private boolean computeQuantiles = true;
+  private boolean computePmf = true;
+  public final transient DefaultInputPort<Double> data = new DefaultInputPort<Double>()
+  {
+    @Override
+    public void process(Double input)
+    {
+
+      quantilesSketch.update(input);
+
+      if (computeQuantiles) {
+        /**
+         * Computes and emits quantiles of the stream seen thus far
+         */
+        quantilesOutput.emit(quantilesSketch.getQuantiles(fractions));
+      }
+
+      if (computeCdf) {
+        /**
+         * Emits (estimate of the) cumulative distribution function evaluated at the input value, according to the
+         * sketched probability distribution of the stream seen thus far.
+         */
+        cdfOutput.emit(quantilesSketch.getCDF(new double[]{input})[0]);
+      }
+
+      if (computePmf) {
+        pmfOutput.emit(quantilesSketch.getPMF(pmfIntervals));
+      }
+
+    }
+  };
+
+  /**
+   * Constructor that allows non-default initialization of the quantile sketch object
+   *
+   * @param k:    Parameter that determines accuracy and memory usage of quantile sketch. See QuantilesSketch documentation
```

Comment:
Agreed. Unfortunately, this is the name that the underlying datasketches implementation uses, with no more descriptive name for this parameter being used either in the datasketches documentation or the research work on which this is based. "sketchSize" would be a reasonable name, but I figured it would be more straightforward for users to look up if I continued to call it k.